### PR TITLE
store/tikv: remove use of GuaranteeLinearizability option in store/tikv

### PR DESCRIPTION
**Pull request by disksing**
Signed-off-by: disksing <i@disksing.com>

### What problem does this PR solve?
Usage of `GuaranteeLinearbility` option in `store/tikv` can be removed.

Part of https://github.com/pingcap/tidb/issues/24302

### What is changed and how it works?
What's Changed:
- Add `SetCasualConsistency` method
- Remove option usages.

### Check List 
Tests 
- Unit test

### Release note 
- No release note

### CODE REVIEW

File: store/tikv/txn.go [hunk 1]
```diff
@@ -283,6 +284,13 @@ func (txn *KVTxn) SetEnable1PC(b bool) {
 	txn.enable1PC = b
 }
 
+// SetCasualConsistency indicates if the transaction does not need to
+// guarantee linearizability. Default value is false which means
+// linearizability is guaranteed.
+func (txn *KVTxn) SetCasualConsistency(b bool) {
```
Comment (ekexium) [inline store/tikv/txn.go:290]:
```suggestion
func (txn *KVTxn) SetCausalConsistency(b bool) {
```

Comment (ekexium) [reply store/tikv/txn.go:290]: There are 2 names which indicate the same thing:`GuaranteeLinearizability` == not `CausalConsistency`. Why don't we use consistent names?

Comment (sticnarf) [reply store/tikv/txn.go:290]: I think the purpose is to let `false` mean linearizability (which is the safer and default option). If the field is uninitialized (thus is `false`), the behavior is default.

Comment (disksing) [reply store/tikv/txn.go]: true. Use default value as the default behaviour is easier to reason correctness.

File: session/session.go [hunk 2]
```diff
@@ -517,7 +517,7 @@ func (s *session) doCommit(ctx context.Context) error {
 	s.txn.SetOption(tikvstore.EnableAsyncCommit, s.GetSessionVars().EnableAsyncCommit)
 	s.txn.SetOption(tikvstore.Enable1PC, s.GetSessionVars().Enable1PC)
 	// priority of the sysvar is lower than `start transaction with causal consistency only`
-	if s.txn.GetOption(tikvstore.GuaranteeLinearizability) == nil {
+	if val := s.txn.GetOption(tikvstore.GuaranteeLinearizability); val == nil || val.(bool) {
```
Comment (sticnarf) [inline session/session.go:520]: `val == nil` seems impossible because `IsCasualConsistency` always returns a bool?

Comment (disksing) [reply session/session.go:520]: There are some other mock store implementations. I guess this way is better as it requires less detail of underlying engine.

**Review by sticnarf — APPROVED** [review]

**Pull request merged**